core/src/main/scala/org/apache/spark/sql/delta/implicits/package.scala

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright (2021) The Delta Lake Project Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.delta
+
+import org.apache.spark.sql.types.{ArrayType, MapType, StructField, StructType}
+
+package object implicits {
+
+  /**
+   * This implicit class is used to provide helpful methods used throughout the code that are not
+   * provided by Spark-Catalyst's StructType.
+   */
+  implicit class RichStructType(structType: StructType) {
+
+    /**
+     * Returns a field in this struct and its child structs, case insensitively.
+     *
+     * If includeCollections is true, this will return fields that are nested in maps and arrays.
+     *
+     * @param fieldNames The path to the field, in order from the root. For example, the column
+     *                   nested.a.b.c would be Seq("nested", "a", "b", "c").
+     */
+    def findNestedFieldIgnoreCase(
```

Comment:
where is this used??

Reply:
Good catch. This will be used in the upcoming Data Skipping PR to resolve issue https://github.com/delta-io/delta/issues/931.

This must have been forgotten during the refactor. I will remove it from this PR.

Reply:
thank you. yeah its important to keep each PR self contained and not have superfluous stuff.